recipes/vtu11/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd, valid_min_cppstd
+from conan.tools.files import get, copy, export_conandata_patches, apply_conandata_patches
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "vtu11"
+    description = "Vtu11 is a small C++ header-only library to write unstructured grids using the vtu file format"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/phmkopp/vtu11"
+    topics = ("vtu", "c++11")
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+    options = {
+        "with_zlib": [True, False],
+    }
+    default_options = {
+        "with_zlib": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.13", transitive_headers=True)
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        if valid_min_cppstd(self, 17):
+            apply_conandata_patches(self)
```

Comment:
this logic is wrong, specially when package id is cleared...

Reply:
Yes, you're correct, good catch. I'm a bit uncertain what the best way is to resolve this. What is your opinion? Should I solve this with a "custom" clear functionality in the package id, clearing everything except settings, compiler version?

Reply:
I would say that the patch should be robust to any C++ standard in the first place.